src/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs

```diff
@@ -955,6 +955,64 @@ public void EndWrite(IAsyncResult asyncResult)
 #endif
         }
 
+        // ReadAsync - provide async read functionality.
+        // 
+        // This method provides async read functionality. All we do is
+        // call through to the Begin/EndRead methods.
+        // 
+        // Input:
+        // 
+        //     buffer            - Buffer to read into.
+        //     offset            - Offset into the buffer where we're to read.
+        //     size              - Number of bytes to read.
+        //     cancellationtoken - Token used to request cancellation of the operation
+        // 
+        // Returns:
+        // 
+        //     A Task<int> representing the read.
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int size, CancellationToken cancellationToken)
+        {
+            if (cancellationToken.CanBeCanceled)
```

Comment:
Good catch; InteliSense completed it for me incorrect, and in my haste I entirely missed it.  It should be IsCancellationRequested, both here and below.